system/lib/libc/dynlink.c

```diff
@@ -356,13 +364,25 @@ int _emscripten_proxy_dlsync_async(pthread_t target_thread, em_promise_t promise
   if (!dlopen_proxying_queue) {
     dlopen_proxying_queue = em_proxying_queue_create();
   }
-  int rtn = emscripten_proxy_async_with_callback(dlopen_proxying_queue,
-                                                 target_thread,
-                                                 do_thread_sync,
-                                                 NULL,
-                                                 done_thread_sync,
-                                                 promise);
-  if (target_thread->sleeping) {
+
+  struct promise_result* info = malloc(sizeof(struct promise_result));
+  if (!info) {
+    return false;
+  }
+  *info = (struct promise_result){.promise = promise, .result = false};
```

Comment:
I think find it little more readable to just do:

```
info->promise = promise;
info->result = false;
```

Is the cast here actually needed?  I guess so?

Reply:
Yeah, it's not actually a cast, but rather part of the [compound literal](https://en.cppreference.com/w/c/language/compound_literal) syntax.

In general I prefer this syntax, but I agree that it's nice to give each field its own line. Maybe I'll try to tweak the clang format settings to make this nicer.

Reply:
It turns out that there's no clang-format setting for this, but putting a comma after the last item achieves the same thing.

Reply:
It just seems like more, and less obvious, syntax than the simple pair of assignments.. but maybe I'm just not familiar with reading code with this syntax.